azure/scope/cluster.go

```diff
@@ -258,18 +266,26 @@ func (s *ClusterScope) LBSpecs() []azure.ResourceSpecGetter {
 	// Node outbound LB
 	if s.NodeOutboundLB() != nil {
 		specs = append(specs, &loadbalancers.LBSpec{
-			Name:                 s.NodeOutboundLB().Name,
-			ResourceGroup:        s.ResourceGroup(),
-			SubscriptionID:       s.SubscriptionID(),
-			ClusterName:          s.ClusterName(),
-			Location:             s.Location(),
-			ExtendedLocation:     s.ExtendedLocation(),
-			VNetName:             s.Vnet().Name,
-			VNetResourceGroup:    s.Vnet().ResourceGroup,
-			FrontendIPConfigs:    s.NodeOutboundLB().FrontendIPs,
-			Type:                 s.NodeOutboundLB().Type,
-			SKU:                  s.NodeOutboundLB().SKU,
-			BackendPoolName:      s.NodeOutboundLB().BackendPool.Name,
+			Name:              s.NodeOutboundLB().Name,
+			ResourceGroup:     s.ResourceGroup(),
+			SubscriptionID:    s.SubscriptionID(),
+			ClusterName:       s.ClusterName(),
+			Location:          s.Location(),
+			ExtendedLocation:  s.ExtendedLocation(),
+			VNetName:          s.Vnet().Name,
+			VNetResourceGroup: s.Vnet().ResourceGroup,
+			FrontendIPConfigs: s.NodeOutboundLB().FrontendIPs,
+			Type:              s.NodeOutboundLB().Type,
+			SKU:               s.NodeOutboundLB().SKU,
+			// @sonasingh46: Ref issue #3253
+			// Change the code to be using only s.NodeOutboundLB().BackendPool.Name in the upcoming
```

Comment:
I don't think we can do that as there is no guarantee that users are upgrading one CAPZ release at a time (and we don't enforce it). What do you think about moving this actually persist the defaulting in the spec, similar to `SetDNSName` https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/fce6cac52c5081ba642c08554e1f4596eae6228b/azure/scope/cluster.go#L945